src/views/List.jsx

```diff
@@ -1,20 +1,43 @@
 import { ListItem } from '../components';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { AddItems } from '../components/AddItems';
 import TextInputElement from '../components/TextInputElement';
+import { comparePurchaseUrgency, urgencyObject } from '../api';
 
 export function List({ data, listPath }) {
 	const [searchItem, setSearchItem] = useState('');
+	const [urgency, setUrgency] = useState(urgencyObject);
+	const listName = listPath.slice(listPath.indexOf('/') + 1);
+
+	useEffect(() => {
+		setUrgency(urgencyObject);
+	}, [urgencyObject]);
+
+	const getClassName = (name) => {
+		const statusArray = Object.entries(urgency).find(([_, items]) => {
+			return Array.from(items).some((item) => item.name === name);
+		});
+		if (!statusArray) {
+			throw new Error(`Failed to get class name of ${name}`);
+		}
+		return statusArray[0];
+	};
```

Comment:
This logic will likely be replaced when we start passing the `urgencyStatus` prop directly to the component, removing the need for managing urgency locally.